snakemake/executors/slurm/slurm_submit.py

```diff
@@ -0,0 +1,349 @@
+            self.format_job_pattern,
+            job=job,
+            properties=job.properties(cluster=self.cluster_params(job)),
+            latency_wait=self.latency_wait,
+            wait_for_files=wait_for_files,
+            path=path,
+            **kwargs,
+        )
+        try:
+            return format_p(pattern)
+        except KeyError as e:
+            raise WorkflowError(
+                "Error formatting jobscript: {} not found\n"
+                "Make sure that your custom jobscript is up to date.".format(e)
+            )
+
+    def write_jobscript(self, job, jobscript, **kwargs):
+        # only force threads if this is not a group job
+        # otherwise we want proper process handling
+        use_threads = "--force-use-threads" if not job.is_group() else ""
+
+        envvars = " ".join(
+            "{}={}".format(var, os.environ[var]) for var in self.workflow.envvars
+        )
+
+        # We need to shorten the exec_job-string, here.
+        # The first part SLURM sees should be the executable, here:
+        # snakemake.
+        to_exec = " ".join(self.exec_job.split(" ")[5:])
+
+        exec_job = self.format_job(
+            to_exec,
+            job,
+            _quote_all=True,
+            use_threads=use_threads,
+            envvars=envvars,
+            **kwargs,
+        )
+        content = self.format_job(self.jobscript, job, exec_job=exec_job, **kwargs)
+
+        logger.debug("Jobscript:\n{}".format(content))
+        with open(jobscript, "w") as f:
+            print(content, file=f)
+
+        os.chmod(jobscript, os.stat(jobscript).st_mode | stat.S_IXUSR | stat.S_IRUSR)
+
+    def run(self, job, callback=None, submit_callback=None, error_callback=None):
+        super()._run(job)
+        workdir = os.getcwd()
+        jobid = job.jobid
+        # generic part of a submission string:
+        # print(self.exec_job)
+        os.makedirs('.snakemake/slurm_logs', exist_ok=True)
+
+        try:
+            call = "sbatch -A {account} -p {partition} \
+                    -J {jobname} \
+                    -o .snakemake/slurm_logs/%x_%j.log \
+                    --export=ALL".format(
+                **job.resources, jobname=self.jobname
+            )
+        except KeyError as e:
+            logger.error(
+                "Missing job submission key '{}' for job '{}'.".format(
+                    e.args[0], job.name
+                )
+            )
+            sys.exit(1)
+        if not job.resources.get('walltime_minutes'):
+            logger.warning("No wall time limit is set, setting 'walltime_minutes' to 1.")
+        call += " -t {walltime_minutes}".format(walltime_minutes = job.resources.get('walltime_minutes', default_value=1))
+        if job.resources.get("constraint"):
+            call += " -C {constraint}".format(**job.resources)
+        # TODO: implement when tempfs-resource is defined
+        # if job.resources.get()
+        if job.resources.get("mem_mb_per_cpu"):
```

Comment:
If mem_mb_per_cpu is really supported explicitly here, we also need to support it for local execution. Sorry, I feel like I asked this several times before (getting old I guess): why do we need it again? Couldn't you do the same computation as [here](https://github.com/snakemake/snakemake/pull/1015/files#diff-0039b8713fcb4087ecfa9fcf6462a997d117e39f12268f7a4694444c1725d8c6R79)?